apps/daimo-mobile/src/action/useCreateAccount.ts

```diff
@@ -0,0 +1,117 @@
+import * as ExpoEnclave from "@daimo/expo-enclave";
+import { useEffect, useState } from "react";
+import { Platform } from "react-native";
+import { Hex } from "viem";
+
+import { useAccount } from "../logic/account";
+import { assert } from "../logic/assert";
+import { trpc } from "../logic/trpc";
+import { ActHandle, SetActStatus, useActStatus } from "./actStatus";
+
+/** Deploys a new contract wallet and registers it under a given username. */
+export function useCreateAccount(name: string): ActHandle {
+  const [as, setAS] = useActStatus();
+
+  // Create enclave key immediately, in the idle state
+  const [key, setKey] = useState<{
+    enclaveKeyName: string;
+    pubKeyHex: Hex | null;
+  }>();
+  useEffect(() => {
+    createEnclaveKey(setAS).then(setKey);
+  }, []);
+
+  // On exec, create contract onchain, claiming name.
+  const result = trpc.deployWallet.useMutation();
+  const exec = async () => {
+    if (!key?.pubKeyHex) return;
+    setAS("loading", "Deploying contract...");
```

Comment:
I guess these logs/comments need to be updated (since no more deploy on this API?)

Reply:
not yet but will soon.

this endpoint should do a single, atomic userOp that
- materializes the account
- infinite approves USDC to the paymaster
- claims the name